### PR TITLE
Updates for warnings from dependencies and CI log optimisations

### LINKED ISSUE #1057: Why do I keep getting stuck here when using AdversarialTrainer ? Where is the problem?
# **here is my error:**
_Precompute adv samples:   0%|          | 0/1 [00:00<?, ?it/s]
Adversarial training epochs:   0%|          | 0/20 [00:00<?, ?it/s]
/home/data/xinyuan/.conda/envs/adv_rob_lxy/lib/python3.7/site-packages/art/estimators/classification/pytorch.py:694: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
  x_grad = torch.tensor(x).to(self._device)
/home/data/xinyuan/.conda/envs/adv_rob_lxy/lib/python3.7/site-packages/art/estimators/classification/pytorch.py:695: UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach() or sourceTensor.clone().detach().requires_grad_(True), rather than torch.tensor(sourceTensor).
  y_grad = torch.tensor(y).to(self._device)_

# here is my code:

### Step 0: Define the neural network model, return logits instead of activation in forward method
_class CNN_model1(nn.Module):
    def __init__(self):
        super(CNN_model1, self).__init__()
        self.conv1=nn.Sequential(
            nn.Conv2d(kernel_size=(3,3), in_channels=3, out_channels=32),
            nn.BatchNorm2d(32), 
            nn.ReLU(),
            nn.MaxPool2d(kernel_size=2)
        )
        self.conv2=nn.Sequential(
            nn.Conv2d(kernel_size=(3,3), in_channels=32, out_channels=64),
            nn.BatchNorm2d(64), 
            nn.ReLU(),
            nn.MaxPool2d(kernel_size=2)
        )
        self.fc1=nn.Sequential(
            nn.Linear(54*54*64, 64),
            nn.ReLU()
        )
        self.fc2=nn.Sequential(
            nn.Linear(64, 2),
        )
    def forward(self, x):
        x=self.conv1(x)
        x=self.conv2(x)
        x=x.contiguous().view(x.size(0), -1)# 扁平化
        x=self.fc1(x)
        x=self.fc2(x)
        return x_

### Step 1: Load the dataset
_train_datagen=ImageDataGenerator(rescale=1./255)
train_loader=train_datagen.flow_from_directory("./dataset/ALL_IDB2_origin" ,
                                                        target_size=(224, 224),
                                                        batch_size=batch_size,
                                                        shuffle=True)

x_train=np.zeros(shape=(nTrain, 224, 224, 3))
y_train=np.zeros(shape=(nTrain, 2))

i=0
for inputs,target in train_loader:
    x_train[i * batch_size : (i + 1) * batch_size] = inputs
    y_train[i * batch_size : (i + 1) * batch_size] = target
    i=i+1
    if i*batch_size>=nTrain:
        break_

### Step 1a: Swap axes to PyTorch's NCHW format

_x_train = np.transpose(x_train, (0, 3, 1, 2)).astype(np.float32)
print(x_train.shape)_

### Step 2: Create the model

_cnn_model1=CNN_model1()_

### Step 2a: Define the loss function and the optimizer

_criterion = nn.CrossEntropyLoss()
optimizer = optim.Adam(cnn_model1.parameters(), lr=0.001)_

### Step 3: Create the ART classifier

_classifier = PyTorchClassifier(
    model=cnn_model1,
    loss=criterion,
    optimizer=optimizer,
    input_shape=(3, 224, 224),
    nb_classes=2,
    device_type='gpu'
)_

### Step 4: Train the ART classifier
_classifier.fit(x_train, y_train, batch_size=10, nb_epochs=10)_

### Step 5: Adversarial Train
_from art.defences.trainer import AdversarialTrainer
from art.attacks.evasion import BasicIterativeMethod

attacks = BasicIterativeMethod(classifier, eps=0.3, eps_step=0.01, max_iter=40)
trainer = AdversarialTrainer(classifier, attacks, ratio=1.0)
trainer.fit(x_train, y_train, nb_epochs=20, batch_size=1)_

### PR DESCRIPTION
**Pull request by beat-buesser**
# Description

This pull request provides updates to address warnings in the CI logs and optimisations of logged messages in CI.

Fixes #1057 

## Type of change

Please check all relevant options.

- [x] Improvement (non-breaking)
- [ ] Bug fix (non-breaking)
- [ ] New feature (non-breaking)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

**Review by minhitbk — COMMENTED** [review]
@beat-buesser : I just have a few questions
- Why don't you set verbose default to False, then you don't need to change every piece of unittests?
- There are a number of places where tests are commented out, why do we need to keep them?

**Comment by beat-buesser** [comment]
Hi @minhitbk Thank you for your review. I haven't set the default for verbose to `False` because I wanted to reduce the size of the logs but not change the default behavior of the attacks for backward compatibility. The commented tests are still valid tests, but have been commented to reduce the CI time when we were running into the time limit on Travis. I think they are kept around for consideration once we move the tests to the new pytest framework.
